### ISSUE TITLE
Rule `firewalld_sshd_port_enabled` fails to remediate on Centos Stream 9

### ISSUE THREAD

**Issue by ggbecker** [ComplianceAsCode/content#9495]
#### Description of problem:
Rule xccdf_org.ssgproject.content_rule_firewalld_sshd_port_enabled fails to remediate because the command used to extract which of the network interface is used by the system returns a different value than the one that the configuration file has.

The shell command probably needs to be improved to extract the correct data in both cases.

Centos Stream 9 results of extracting the network interface information (the system has a file `/etc/NetworkManager/system-connections/ens3.nmconnection`:

```
[root@ci-vm-... ~]# ip link show up | cut -d ' ' -f2 | cut -d ':' -s -f1 | grep -E '^(en|eth)'
eth0
[root@ci-vm-... ~]# ip link show up
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP mode DEFAULT group default qlen 1000
    link/ether fa:16:3e:92:6f:a1 brd ff:ff:ff:ff:ff:ff
    altname enp0s3
    altname ens3
```

Fedora 36:
```
~ 0 $ ip link show up | cut -d ' ' -f2 | cut -d ':' -s -f1 | grep -E '^(en|eth)'
enp0s31f6
~ 0 $ ip link show up
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: enp0s31f6: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc fq_codel state DOWN mode DEFAULT group default qlen 1000
    link/ether 38:f3:ab:45:08:1e brd ff:ff:ff:ff:ff:ff
4: wlp0s20f3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DORMANT group default qlen 1000
    link/ether 4c:79:6e:71:d0:13 brd ff:ff:ff:ff:ff:ff
5: virbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default qlen 1000
    link/ether 52:54:00:29:d8:ed brd ff:ff:ff:ff:ff:ff
31: tap1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel master virbr0 state UNKNOWN mode DEFAULT group default qlen 1000
    link/ether fe:5b:4a:c2:b1:c6 brd ff:ff:ff:ff:ff:ff
49: tun0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1360 qdisc fq_codel state UNKNOWN mode DEFAULT group default qlen 500
    link/none 
73: tap0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel master virbr0 state UNKNOWN mode DEFAULT group default qlen 1000
    link/ether fe:7a:ad:67:35:be brd ff:ff:ff:ff:ff:ff
```

Results from a upstream test containing HTML reports and SOS report and files from the testing machine can be inspected: https://artifacts.dev.testing-farm.io/281d7ef6-d3f3-47c3-9ffd-57b5f7764640/work-stigRa4hR3/tests/fmf-plans/stig/execute/data/Sanity/machine-hardening/stig/data/

**Comment by yuumasato** [comment]
I'm not sure what is going on with NetworkManager on the Centos Stream 9 in Testing Farm.
Looking at results from https://artifacts.dev.testing-farm.io/0c03cf94-386d-48ec-9dd4-a0aa3959dce2/ the NM profile available doesn't match the available links.

```
+ ls /etc/NetworkManager/system-connections/
ens3.nmconnection
+ ip link show up
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether 06:9d:8f:1b:b1:34 brd ff:ff:ff:ff:ff:ff
    altname enp0s5
    altname ens5
```

**Comment by marcusburghardt** [comment]
I looked this rule now and there are more issues with the Bash remediation:
* It is not necessary to copy the zone file to /etc/firewalld in this case.
* The remediation assumes the NetworkManager connection name is the same of the interface name.
  * This is common in most of the cases but is not a rule. The NetworkManager connection name could be any other name.
* The remediation assumes the "first" interface listed by the `ip link show up` is the correct one to be used. This is not reliable.
* `nmcli` and `firewall-cmd` commands can be used to make this remediation much more simple and precise than checking xml files.

About the Ansible remediation, it is very different from the Bash remediation. They should be aligned.

In addition, the OVAL assessment is confusing. There are cases where the NIC is checked but in other similar cases it is not checked. Probably there is space for improvements here as well.

So, in short, this entire rule needs to be reviewed and improved. I am happy to assume this issue and send a PR for improvements in this rule.

**Comment by ggbecker** [comment]
Related to: https://bugzilla.redhat.com/show_bug.cgi?id=2116474
